### PR TITLE
[HttpClient] throw exception when AsyncDecoratorTrait gets an already consumed response

### LINKED ISSUE #40639: [HttpClient] A chunk passthru yields null instead of ChunkInterface instances
**symfony/http-client version**: 5.2.4
**symfony/messenger version**: 5.2.4
**symfony/amazon-sqs-messenger version**: 5.2.4

**Description**
I'm receiving `A chunk passthru must yield instances of "Symfony\Contracts\HttpClient\ChunkInterface", "null" yielded` error on Symfony messenger worker container, which handles messages from Amazon SQS queue:

```
Fatal error: Uncaught LogicException: A chunk passthru must yield instances of "Symfony\Contracts\HttpClient\ChunkInterface", "null" yielded. in /app/vendor/symfony/http-client/Response/AsyncResponse.php:349

Stack trace:
#0 /app/vendor/symfony/http-client/Response/AsyncResponse.php(320): Symfony\Component\HttpClient\Response\AsyncResponse::passthruStream(Object(Symfony\Component\HttpClient\Response\CurlResponse), Object(Symfony\Component\HttpClient\Response\AsyncResponse), NULL, Object(SplObjectStorage)) | 
#1 /app/vendor/symfony/http-client/Response/AsyncResponse.php(275): Symfony\Component\HttpClient\Response\AsyncResponse::passthru(Object(Symfony\Component\HttpClient\CurlHttpClient), Object(Symfony\Component\HttpClient\Response\AsyncResponse), Object(Symfony\Component\HttpClient\Chunk\LastChunk), Object(SplObjectStorage)) |  
#2 /app/vendor/symfony/http-client/Response/AsyncResponse.php(63): Symfony\Component\HttpClient\Response\AsyncResponse::stream(Array) |  
#3 /app/vendor/symfony/http-client/Response/CommonResponseTrait.php(1 in /app/vendor/symfony/http-client/Response/AsyncResponse.php on line 349
```

**Additional context**
The `AsyncResponse::passthru` (line 275) is getting `Symfony\Component\HttpClient\Chunk\LastChunk` as third argument. Shouldn't this be denied by https://github.com/symfony/http-client/blob/5.x/Response/AsyncResponse.php#L309?

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40639
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/HttpClient/RetryableHttpClient.php [hunk 1]
```diff
@@ -77,14 +77,17 @@ public function request(string $method, string $url, array $options = []): Respo
...
                         if (null !== $firstChunk) {
                             yield $firstChunk;
-                            yield $context->createChunk($content);
-                            yield $chunk;
-                        } else {
-                            yield $chunk;
                         }
-                        $content = '';
+
+                        while ('' !== $c = $content) {
```
Comment (stof) [inline src/Symfony/Component/HttpClient/RetryableHttpClient.php]: isn't this just a `if ($content !== '')` ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/RetryableHttpClient.php]: I reverted this change which was part of a previous attempt at fixing the issue.

File: src/Symfony/Component/HttpClient/Response/AsyncResponse.php [hunk 2]
```diff
@@ -272,6 +272,14 @@ public static function stream(iterable $responses, float $timeout = null, string
                     continue;
                 }
 
+                if (null !== $chunk->getError()) {
+                    // no-op
+                } elseif ($chunk->isFirst()) {
+                    $r->yieldedState = 1;
```
Comment (stof) [inline src/Symfony/Component/HttpClient/Response/AsyncResponse.php]: should we use private constants for those states to make it easier to understand what they represent ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/Response/AsyncResponse.php]: ok, updated

**Pull request merged**